VocaDbWeb/Views/Event/Edit.cshtml

```diff
@@ -179,11 +179,19 @@
 				@Html.HiddenForKnockout(m => m.SongList, "value: ko.toJSON(songList.entry())")
 			</div>
 	
+			<div class="editor-label">
+				Venue
+			</div>
+			<div class="editor-field">
+				@KnockoutHelpers.LockingAutoComplete("venueAutoComplete", "venue.entry", "venue.name")
+				@Html.HiddenForKnockout(m => m.Venue, "value: ko.toJSON(venue)")
+			</div>
+
 			<div class="editor-label">
 				@Helpers.HelpLabel("Venue", "Can be either a physical location such as concert hall, or a virtual location (website).")
 			</div>
 			<div class="editor-field">
-				@Html.TextBoxFor(m => m.Venue, new { size = 40, @class = "span3", maxlength = 1000 })<br />
+				@Html.TextBoxFor(m => m.VenueName, new { size = 40, @class = "span3", maxlength = 1000 })<br />

```

Comment:
So you allow entering both venue link and venue name? I think that, if venue is selected from the autocomplete, then we can hide this venue name field and automatically fill it based on the selected venue. 

The venue name field can be used when there is no entry for the venue, if it's a one-off event in that venue. This should also be explained in the description.